### ISSUE TITLE
Proposed addition of “price” to “terms” during an etch

### ISSUE THREAD

**Issue by encloinc** [ordinals/ord#3794]
Hey! I wanted to know if we were to submit a PR to ord altering the indexer to do the following if it’s something that has a possibility of being approved.

## The addition of a “price” field
Currently an etches “terms” looks like the following:

struct Terms {
  amount: Option<u128>,
  cap: Option<u128>,
  height: (Option<u64>, Option<u64>),
  offset: (Option<u64>, Option<u64>),
}

I propose an OPTIONAL extra field “price” to be added to these terms, expressed as the following:
`price: Option<u128>`

## How the “price” field would function
Price is an integer expressed in Satoshi. During an etch, IF the price field is present the following would be added as a requirement:

#### The address owning the first NON-OP_RETURN transaction output of an Etch transaction or, in the case it’s provided, the address owning the transaction output specified by “pointer” of an Etch transaction:

A transaction including a runestone that is minting MUST include a transaction output transferring a MINIMUM of “price” amount of satoshi to ”Etch Output”

If there are no transaction outputs in vout that transfer a MINIMUM of “price” to “Etch Output”, the mint is ignored BUT the transaction is not considered a cenotaph and edicts are still processed.

If no price field is provided, all functionality above is ignored.

## Usecase
This would create a decentralized way to create ICOs natively on Bitcoin - allowing the fungibility of Rune to be used as a mechanism to fund innovative projects on the space

## Final thoughts
I leave my proposal here for feedback and if there’s interest / probability of merging or being added, I would be happy to implement in ord through a PR.

Thanks for reading :)

**Comment by ayo-dan** [comment]
This would be a great addition to the protocol.

There are massive risks when it comes to token sales but the mechanism described is much more efficient than a project premining and then selling their runes.

It could be worth checking out [OrdinalsBot's approach to rune etching/minting](https://ordinalsbot.com/launchpad). When setting the details of your Rune you can set a price and revenue address.

**Comment by encloinc** [comment]
Also had the idea of adding an asset field, which takes in a RuneID. That way, mint can be priced in other tunes rather than satoshi.

this would have other usecase a too, for example, wrapped runes that represent other assets (like BTC, USD) which can only be minted by a custodian that has the asset that it costs to mint the wrapped rune (because they would be the only ones that have it)

Runes could even be used to stable coin natively on Bitcoin with this mechanism, with institutions (like tether) being the only ones authorized to mint since they are the only holder of the rune described in RuneID (assuming they premined that rune). 

TLDR: granular access to a runes mint function could be granted by leveraging the fungibility of the Runes protocol.
